cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -245,15 +245,27 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
 			}
 		}
 		if containsInBindingName != "" {
-			targetPropertyValue := springBootProject.applicationProperties[targetPropertyName]
+			// get distinct container names
+			var containerNames []string
+			seen := make(map[string]struct{})
+			for key, value := range springBootProject.applicationProperties {
+				if strings.HasSuffix(key, targetPropertyName) {
```

Comment:
Same here: target property name need to be updated. Not only update the string, need create some func like `getBindingDestinationMap`.

Reply:
Same as above.